### PR TITLE
Add support for custom authorization header value

### PR DESCRIPTION
**Pull request by DaniilShmelev**
typed-rest-client doesn't support `Authorization: token ...` by default, so we need to pass it as an HTTP header.
This is needed for python release binary downloading from the [python versions registry](https://github.com/actions/python-versions).

### CODE REVIEW

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by alexander-smolyakov — APPROVED** [review]

File: tool.ts [hunk 1]
```diff
@@ -194,11 +194,17 @@ export function findLocalToolVersions(toolName: string, arch?: string) {
 /**
  * Download a tool from an url and stream it into a file
  *
- * @param url       url of tool to download
- * @param fileName  optional fileName.  Should typically not use (will be a guid for reliability). Can pass fileName with an absolute path.
- * @param handlers  optional handlers array.  Auth handlers to pass to the HttpClient for the tool download.
+ * @param url                url of tool to download
+ * @param fileName           optional fileName.  Should typically not use (will be a guid for reliability). Can pass fileName with an absolute path.
+ * @param handlers           optional handlers array.  Auth handlers to pass to the HttpClient for the tool download.
+ * @param additionalHeaders  optional custom HTTPheaders.  This is passed to the download client.
```
Comment (anatolybolshakov) [inline tool.ts:200]: What is the 'download client' here? Could you please elaborate it more in comments?

**Review by EzzhevNikita — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "azure-pipelines-tool-lib",
-  "version": "1.1.0",
+  "version": "1.1.1",
```
[the remark this answers is not in the document]
Comment (DaniilShmelev) [reply package.json]: Yeah, makes sense

File: tool.ts (hunk 1 above)
Comment (DaniilShmelev) [reply tool.ts]: Cleared it up

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

**Comment by DaniilShmelev** [comment]
@mmrazik Could you please review and merge this PR?
Also, could you publish the package?

**Comment by DaniilShmelev** [comment]
@mmrazik hi, could you please the package to npm?

**Pull request merged**